### PR TITLE
Added tables to replace the introductory part on core properties - bis for github

### PR DESCRIPTION
**Pull request by iherman**
resolving merge conflicts in #528 would have been terrible; git's rebase went completely crazy. However, my changes in #528 were fully concentrated to the initial part of §5, so I decided to make a fresh branch and manually transfer my changes instead.

@rhiaro, would you mind checking that the changes here are indeed the same as the changes proposed in #528 (except that they are on the fresh branch)? If so, @msporny, I believe the comments in #528 stand for this one, and this PR could be merged (and #528 simply closed).

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/did-core/pull/536.html" title="Last updated on Jan 11, 2021, 1:35 PM UTC (41af738)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/did-core/536/c8bc6b1...41af738.html" title="Last updated on Jan 11, 2021, 1:35 PM UTC (41af738)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1320,63 +1320,200 @@ <h1>Core Properties</h1>
...
+              <td><code>type</code></td>
+              <td>REQUIRED</td>
+              <td>
+                A <a data-cite="INFRA#string">string</a> or an <a data-cite="INFRA#ordered-set">ordered set</a> of <a data-cite="INFRA#string">strings</a>.
+              </td>
+            </tr>
+            <tr>
+              <td><code><a>publicKeyJwk</a></code></td>
+              <td>OPTIONAL</td>
+              <td>A <a data-cite="INFRA#maps">map</a> representing a JSON Web Key that conforms to [[RFC7517]].</td>
```
Comment (rhiaro) [inline index.html:1461]:
```suggestion
              <td>A <a data-cite="INFRA#maps">map</a> representing a JSON Web Key that conforms to [[RFC7517]]. This value MUST NOT contain "d", or any other members of the private information class as described in <a href="https://tools.ietf.org/html/rfc7517#section-8.1.1">Registration Template</a>.</td>
```

File: index.html [hunk 2]
```diff
@@ -1320,63 +1320,200 @@ <h1>Core Properties</h1>
...
+              <td>OPTIONAL</td>
+              <td>
+                An <a data-cite="INFRA#ordered-set">ordered set</a> of either <a>Verification Method</a> <a data-cite="INFRA#ordered-map">maps</a> or <a data-cite="INFRA#string">strings</a> that conform to the rules of [[RFC3986]] for <a>URI</a>-s. <br/>The <a>URI</a> values SHOULD also conform to the rules in Section <a href="#did-url-syntax"></a>.
+              </td>
+            </tr>
+            <tr>
+              <td><code><a>assertionMethod</a></code></td>
+              <td>OPTIONAL</td>
+              <td>
+                An <a data-cite="INFRA#ordered-set">ordered set</a> of either <a>Verification Method</a> <a data-cite="INFRA#ordered-map">maps</a> or <a data-cite="INFRA#string">strings</a> that conform to the rules of [[RFC3986]] for <a>URI</a>-s. <br/>The <a>URI</a> values SHOULD also conform to the rules in Section <a href="#did-url-syntax"></a>.
```
Comment (rhiaro) [inline index.html:1392]:
```suggestion
                An <a data-cite="INFRA#ordered-set">ordered set</a> of either <a>Verification Method</a> <a data-cite="INFRA#ordered-map">maps</a> (see <a href="#core-properties-for-a-verification-method"></a>) or <a data-cite="INFRA#string">strings</a> that conform to the rules of [[RFC3986]] for <a>URIs</a>. <br/>The <a>URI</a> values SHOULD also conform to the rules in Section <a href="#did-url-syntax"></a>.
```

File: index.html [hunk 3]
```diff
@@ -1320,63 +1320,200 @@ <h1>Core Properties</h1>
...
+              </td>
+            </tr>
+            <tr>
+              <td><code><a>keyAgreement</a></code></td>
+              <td>OPTIONAL</td>
+              <td>
+                An <a data-cite="INFRA#ordered-set">ordered set</a> of either <a>Verification Method</a> <a data-cite="INFRA#ordered-map">maps</a> or <a data-cite="INFRA#string">strings</a> that conform to the rules of [[RFC3986]] for <a>URI</a>-s. <br/>The <a>URI</a> values SHOULD also conform to the rules in Section <a href="#did-url-syntax"></a>.
```
Comment (rhiaro) [inline index.html:1399]:
```suggestion
                An <a data-cite="INFRA#ordered-set">ordered set</a> of either <a>Verification Method</a> <a data-cite="INFRA#ordered-map">maps</a> (see <a href="#core-properties-for-a-verification-method"></a>) or <a data-cite="INFRA#string">strings</a> that conform to the rules of [[RFC3986]] for <a>URIs</a>. <br/>The <a>URI</a> values SHOULD also conform to the rules in Section <a href="#did-url-syntax"></a>.
```

File: index.html [hunk 4]
```diff
@@ -1320,63 +1320,200 @@ <h1>Core Properties</h1>
...
+              </td>
+            </tr>
+            <tr>
+              <td><code><a>capabilityInvocation</a></code></td>
+              <td>OPTIONAL</td>
+              <td>
+                An <a data-cite="INFRA#ordered-set">ordered set</a> of either <a>Verification Method</a> ins<a data-cite="INFRA#ordered-map">maps</a>ances or <a data-cite="INFRA#string">strings</a> that conform to the rules of [[RFC3986]] for <a>URI</a>-s. <br/>The <a>URI</a> values SHOULD also conform to the rules in Section <a href="#did-url-syntax"></a>.
```
Comment (rhiaro) [inline index.html:1406]:
```suggestion
                An <a data-cite="INFRA#ordered-set">ordered set</a> of either <a>Verification Method</a> <a data-cite="INFRA#ordered-map">maps</a> (see <a href="#core-properties-for-a-verification-method"></a>) or <a data-cite="INFRA#string">strings</a> that conform to the rules of [[RFC3986]] for <a>URIs</a>. <br/>The <a>URI</a> values SHOULD also conform to the rules in Section <a href="#did-url-syntax"></a>.
```

File: index.html [hunk 5]
```diff
@@ -1320,63 +1320,200 @@ <h1>Core Properties</h1>
...
+              </td>
+            </tr>
+            <tr>
+              <td><code><a>capabilityDelegation</a></code></td>
+              <td>OPTIONAL</td>
+              <td>
+                An <a data-cite="INFRA#ordered-set">ordered set</a> of either <a>Verification Method</a> <a data-cite="INFRA#ordered-map">maps</a> or <a data-cite="INFRA#string">strings</a> that conform to the rules of [[RFC3986]] for <a>URI</a>-s. <br/>The <a>URI</a> values SHOULD also conform to the rules in Section <a href="#did-url-syntax"></a>.
```
Comment (rhiaro) [inline index.html:1413]:
```suggestion
                An <a data-cite="INFRA#ordered-set">ordered set</a> of either <a>Verification Method</a> <a data-cite="INFRA#ordered-map">maps</a> (see <a href="#core-properties-for-a-verification-method"></a>) or <a data-cite="INFRA#string">strings</a> that conform to the rules of [[RFC3986]] for <a>URIs</a>. <br/>The <a>URI</a> values SHOULD also conform to the rules in Section <a href="#did-url-syntax"></a>.
```

File: index.html [hunk 6]
```diff
@@ -1320,63 +1320,200 @@ <h1>Core Properties</h1>
...
+              </td>
+            </tr>
+            <tr>
+              <td><code><a>service</a></code></td>
+              <td>OPTIONAL</td>
+              <td>An <a data-cite="INFRA#ordered-set">ordered set</a> of <a>Service Endpoint</a> <a data-cite="INFRA#ordered-map">maps</a>.</td>
```
Comment (rhiaro) [inline index.html:1419]:
```suggestion
              <td>An <a data-cite="INFRA#ordered-set">ordered set</a> of <a>Service Endpoint</a> <a data-cite="INFRA#ordered-map">maps</a> (see <a href="#core-properties-for-a-service-endpoint"></a>).</td>
```

File: index.html [hunk 7]
```diff
@@ -1320,63 +1320,200 @@ <h1>Core Properties</h1>
...
+              <th>Usage constraints</th>
+              <th>Value constraints</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <td><code>id</code></td>
+              <td>REQUIRED</td>
+              <td>
+                A <a data-cite="INFRA#string">string</a> that conforms to the rules of [[RFC3986]] for <a>URI</a>-s and which SHOULD also conform to the rules in Section <a href="#did-url-syntax"></a>.
```
Comment (rhiaro) [inline index.html:1441]:
```suggestion
                A <a data-cite="INFRA#string">string</a> that conforms to the rules of [[RFC3986]] for <a>URIs</a> and which SHOULD also conform to the rules in Section <a href="#did-url-syntax"></a>.
```

File: index.html [hunk 8]
```diff
@@ -1320,63 +1320,200 @@ <h1>Core Properties</h1>
...
+              <th>Usage constraints</th>
+              <th>Value constraints</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <td><code>id</code></td>
+              <td>REQUIRED</td>
+              <td>
+                when used on a <a>Service Endpoint</a>: a <a data-cite="INFRA#string">string</a> that conforms to the rules of [[RFC3986]] for <a>URI</a>-s.
```
Comment (rhiaro) [inline index.html:1492]:
```suggestion
                a <a data-cite="INFRA#string">string</a> that conforms to the rules of [[RFC3986]] for <a>URIs</a>.
```

I think the "when used on a service endpoint" part is implicit by the table

File: index.html [hunk 9]
```diff
@@ -1320,63 +1320,200 @@ <h1>Core Properties</h1>
...
+              <td>REQUIRED</td>
+              <td>
+                A <a data-cite="INFRA#string">string</a> or an <a data-cite="INFRA#ordered-set">ordered set</a> of <a data-cite="INFRA#string">strings</a>.
+              </td>
+            </tr>
+            <tr>
+              <td><code><a>serviceEndpoint</a></code></td>
+              <td>OPTIONAL</td>
+              <td>
+                A <a data-cite="INFRA#string">string</a> that conform to the rules of [[RFC3986]] for <a>URI</a>-s, a <a data-cite="INFRA#string">map</a>, or an <a data-cite="INFRA#ordered-set">ordered set</a> composed of a one or more <a data-cite="INFRA#string">strings</a> that conform to the rules of [[RFC3986]] for <a>URI</a>-s and/or <a data-cite="INFRA#string">maps</a>.
```
Comment (rhiaro) [inline index.html:1506]:
```suggestion
                A <a data-cite="INFRA#string">string</a> that conform to the rules of [[RFC3986]] for <a>URIs</a>, a <a data-cite="INFRA#string">map</a>, or an <a data-cite="INFRA#ordered-set">ordered set</a> composed of a one or more <a data-cite="INFRA#string">strings</a> that conform to the rules of [[RFC3986]] for <a>URIs</a> and/or <a data-cite="INFRA#string">maps</a>.
```

**Review by rhiaro — CHANGES_REQUESTED** [review]
I've said it before, but this arrangement of the core properties is so much better and easier to read!

My one query is - do Verification Methods need to be referred to by DIDs (as stated here - but not in current spec text) or can they be any URIs?

And the caveat that once this is merged I will clean up the rest of the prose in the section so normative statements aren't duplicated. I also think we should use `<dfn>` tags around the properties in the table, but that can come alongside the prose cleanup.

**Comment by iherman** [comment]
@rhiaro thanks for the editorial comments; I have taken care of them.

As for

> My one query is - do Verification Methods need to be referred to by DIDs (as stated here - but not in current spec text) or can they be any URIs?

yes, I was wondering about that myself. _All_ examples in the text use DID URL-s for the various verifications methods but the spec is silent on this aspect altogether. This is one of the places when the reader might be influenced by the examples.

I have put a _SHOULD_ for a DID URL to reflect the examples. If that is incorrect, then at least one examples, if not several, should use a different URI to make the point. Personally, I can live with either way.

**Comment by iherman** [comment]
(In case I was not clear: https://github.com/w3c/did-core/pull/536/commits/41af738c47a4f612f61a9c2388bea1fc45249175 covers the editorial changes.)

**Comment by rhiaro** [comment]
> I have put a SHOULD for a DID URL to reflect the examples.

I checked with @msporny about this earlier, and I think the SHOULD here is correct.

**Comment by iherman** [comment]
@msporny can we/you merge this PR (and close #528)?

**Review by rhiaro — APPROVED** [review]

**Review by msporny — APPROVED** [review]

**Comment by msporny** [comment]
Editorial, multiple reviews here and in #528, changes requested and made, no objections, merging.

**Pull request merged**